mitiq/qse/qse_utils.py

```diff
@@ -48,24 +47,33 @@ def get_projector(
     return projector
 
 
-def _get_expectation_value_for_observable(
+def get_expectation_value_for_observable(
     circuit: QPROGRAM,
-    executor: Callable[[QPROGRAM], QuantumResult],
+    executor: Union[Executor, Callable[[QPROGRAM], QuantumResult]],
     observable: Union[PauliString, Observable],
     pauli_string_to_expectation_cache: Dict[PauliString, complex] = {},
 ) -> float:
+    """Provide pauli_string_to_expectation_cache if you want to take advantage
+    of caching.
+
+    This function modifies pauli_string_to_expectation_cache in place.
+    """
+
     def get_expectation_value_for_one_pauli(
         pauli_string: PauliString,
     ) -> float:
         cache_key = pauli_string.with_coeff(1)
-        pauli_string_to_expectation_cache[cache_key] = Observable(
-            cache_key
-        ).expectation(circuit, executor)
+        pauli_string_to_expectation_cache[cache_key] = final_executor.evaluate(
+            circuit, Observable(cache_key)
+        )[0]
         return (
             pauli_string_to_expectation_cache[cache_key] * pauli_string.coeff
         ).real
 
     total_expectation_value_for_observable = 0.0
+    final_executor: Executor = (
```

Comment:
initially i was overriding the variable, but pylint was getting confused by the type of `executor` in line 66, so I had to change the name.